apps/wallet/frontend/src/components/ListAllocationRequests.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { Card, CardContent, Chip, Stack } from '@mui/material';
+import Typography from '@mui/material/Typography';
+import { useTokenStandardAllocationRequests } from '../hooks/useTokenStandardAllocationRequests';
+import { AmountDisplay, Loading } from '@lfdecentralizedtrust/splice-common-frontend';
+import { AllocationRequest } from '@daml.js/splice-api-token-transfer-instruction/lib/Splice/Api/Token/AllocationRequestV1/module';
+import { Contract } from '@lfdecentralizedtrust/splice-common-frontend-utils';
+import { usePrimaryParty } from '../hooks';
+import dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import BigNumber from 'bignumber.js';
+import BftAnsEntry from './BftAnsEntry';
+import { ArrowCircleLeftOutlined } from '@mui/icons-material';
+
+dayjs.extend(relativeTime);
+
+const ListAllocationRequests: React.FC = () => {
+  const allocationRequestsQuery = useTokenStandardAllocationRequests();
+  const primaryPartyId = usePrimaryParty();
+
+  if (allocationRequestsQuery.isLoading || !primaryPartyId) {
+    return <Loading />;
+  }
+  if (allocationRequestsQuery.isError) {
+    return (
+      <Typography color="error">
+        Error loading allocation requests: {JSON.stringify(allocationRequestsQuery.error)}
+      </Typography>
+    );
+  }
+
+  const allocationRequests = allocationRequestsQuery.data || [];
+
+  return (
+    <Stack spacing={4} direction="column" justifyContent="center" id="allocation-requests">
+      <Typography mt={6} variant="h4">
+        Allocation Requests <Chip label={allocationRequests.length} color="success" />
+      </Typography>
+      {allocationRequests.map(ar => (
+        <AllocationRequestDisplay key={ar.contractId} request={ar} userParty={primaryPartyId} />
+      ))}
+    </Stack>
+  );
+};
+
+const AllocationRequestDisplay: React.FC<{
+  request: Contract<AllocationRequest>;
+  userParty: string;
+}> = ({ request, userParty }) => {
+  const { transferLegs, settlement } = request.payload;
+
+  const transferLegId = Object.keys(transferLegs).find(
+    transferLegId => transferLegs[transferLegId].sender === userParty
```

Comment:
Supporting a single transfer-leg only won't work e.g. for splitwell, where one settlement requires the sender to send money to all the counter-parties that they owe money to.

I'd suggest the following display for allocation requests:
1. list settlementinfo at the top, in a way that makes it easy to identify the trade being settled.
2. list all transfer legs in a table below the settlement info, one column per record field.
3. add one extra column to this table where you show the actions available on this leg:
  - if there's no matching allocation: (accept) button
  - if there is a mtaching allocation: "allocated <date>" (withdraw) button to withdraw the allocation

Reply:
Note that showing all legs also makes the request easier to identify and trust, as one will see the counter-parties' legs as well.